golang/trace/zipin_tracereporter_test.go

```diff
@@ -0,0 +1,190 @@
+			Port:        8888,
+			ServiceName: "test",
+		},
+		Name:        name,
+		Id:          uInt64ToBytes(span.SpanID()),
+		ParentId:    uInt64ToBytes(span.ParentID()),
+		Annotations: buildZipkinAnnotations(annotations),
+		Debug:       false,
+	}
+	reflect.DeepEqual(thriftSpan, expectedSpan)
+}
+
+func TestInetAton(t *testing.T) {
+	assert.Equal(t, inetAton("1.0.0.1"), (uint32)(16777217))
+}
+
+func TestUInt64ToBytes(t *testing.T) {
+	assert.Equal(t, uInt64ToBytes(54613478251749257), []byte("\x00\xc2\x06\xabK$\xdf\x89"))
+}
+
+func TestBase64Encode(t *testing.T) {
+	assert.Equal(t, base64Encode(12711515087145684), "AC0pDj1TitQ=")
+}
+
+func RandomAnnotations() []tchannel.Annotation {
+	anns := make([]tchannel.Annotation, 4)
+	anns[0] = tchannel.Annotation{
+		Key:       tchannel.AnnotationKeyClientReceive,
+		Timestamp: time.Now(),
+	}
+	anns[1] = tchannel.Annotation{
+		Key:       tchannel.AnnotationKeyClientSend,
+		Timestamp: time.Now(),
+	}
+	anns[2] = tchannel.Annotation{
+		Key:       tchannel.AnnotationKeyServerReceive,
+		Timestamp: time.Now(),
+	}
+	anns[3] = tchannel.Annotation{
+		Key:       tchannel.AnnotationKeyServerSend,
+		Timestamp: time.Now(),
+	}
+
+	return anns
+}
+
+type testArgs struct {
+	s *mocks.TChanTCollector
+	c gen.TChanTCollector // ?
+}
+
+func ctxArg() mock.AnythingOfTypeArgument {
+	return mock.AnythingOfType("*tchannel.headerCtx")
+}
+
+func TestSubmit(t *testing.T) {
+	withSetup(t, func(ctx thrift.Context, args testArgs) {
+		host := &tchannel.Endpoint{
+			Ipv4:        "127.0.0.1",
+			Port:        8888,
+			ServiceName: "test",
+		}
+		span := *tchannel.NewRootSpan()
+		name := "test"
+		annotations := RandomAnnotations()
+		thriftSpan := buildZipkinSpan(span, annotations, nil, name, host)
+		thriftSpan.BinaryAnnotations = []*gen.BinaryAnnotation{}
+		ret := &gen.Response{Ok: true}
+		args.s.On("Submit", ctxArg(), thriftSpan).Return(ret, nil)
+		got, err := args.c.Submit(ctx, thriftSpan)
+		require.NoError(t, err)
+		assert.Equal(t, ret, got)
+	})
+}
+
+func withSetup(t *testing.T, f func(ctx thrift.Context, args testArgs)) {
+	args := testArgs{
+		s: new(mocks.TChanTCollector),
+	}
+
+	ctx, cancel := thrift.NewContext(time.Second * 10)
+	defer cancel()
+
+	// Start server
+	tchan, listener, err := setupServer(args.s)
+	require.NoError(t, err)
+	defer tchan.Close()
+
+	// Get client1
+	args.c, err = getClient(listener.Addr().String())
+	require.NoError(t, err)
+
+	f(ctx, args)
+
+	args.s.AssertExpectations(t)
+}
+
+func setupServer(h *mocks.TChanTCollector) (*tchannel.Channel, net.Listener, error) {
+	tchan, err := tchannel.NewChannel("service", nil)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	listener, err := net.Listen("tcp", ":0")
+	if err != nil {
+		return nil, nil, err
+	}
+
+	server := thrift.NewServer(tchan)
+	server.Register(gen.NewTChanTCollectorServer(h))
+
+	tchan.Serve(listener)
+	return tchan, listener, nil
+}
+
+func getClient(dst string) (gen.TChanTCollector, error) {
```

Comment:
You don't want to use the Thrift client here - you're just testing the Thrift logic instead of you ZipkinTraceReporter. You need to make your client a ZipkinTraceReporter that you call Submit on and make sure it's calling the server with the expected arguments.